### PR TITLE
Add SEO related meta tags for website

### PR DESCRIPTION
**Pull request by IsAmrish**
Signed-off-by: isamrish <askmaurya48@gmail.com>

### CODE REVIEW

File: website/src/pages/Blog/index.tsx [hunk 1]
```diff
@@ -190,6 +193,7 @@ const Blog: React.FC = () => {
 
   return (
     <>
+     <Metadata title={SeoJson.pages.blog.title} description={SeoJson.pages.blog.description} url={`${getHostOrigin}${SeoJson.pages.blog.url}`} image={`${getHostOrigin}${SeoJson.pages.blog.image}`} isPost={false} type="Series"  />
```
Comment (Pallavi-PH) [inline website/src/pages/Blog/index.tsx:196]:
```suggestion
     <Metadata title={SeoJson.pages.blog.title} description={SeoJson.pages.blog.description} url={`${getHostOrigin}${SeoJson.pages.blog.url}`} image={`${getHostOrigin}${SeoJson.pages.blog.image}`} isPost={false} type="Article"  />
```
Type should be article right?

Comment (IsAmrish) [reply website/src/pages/Blog/index.tsx:196]: A `Series` in schema.org is a group of related items
https://schema.org/Series

**Comment by rakeshPRaghu** [comment]
can we remove the unwanted line breaks and extra spaces!?

File: website/src/components/Metadata/index.tsx [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+    const jsonLd = isPost ? articleSchema({title, description, url, image: imageObj, author, tags, site }) : websiteSchema( {title, description, url, image: imageObj, type , site });
+    return(
+        <Helmet>
+            <title>{title || defaultConfig.title}</title>
+            <meta name="description" content={description || defaultConfig.description} />
+            <meta name="image" content={image} />
+            <link rel="canonical" href={url || defaultConfig.url} />
+             {/** Open Graph  */}
+            <meta property="og:type" content={isPost ? 'article' : 'website'} />
```
[the remark this answers is not in the document]
Comment (IsAmrish) [reply website/src/components/Metadata/index.tsx:51]: thanks, it was missing :)
added it!

File: website/src/components/Metadata/index.tsx [hunk 3]
```diff
@@ -0,0 +1,82 @@
+import React from "react";
+import {Helmet} from "react-helmet";
+import { getHostOrigin } from "../../utils/getHostOrigin";
+import { articleSchema } from "./articleSchema";
+import { websiteSchema } from "./websiteSchema";
+import { Author } from "./metadata.models";
+
+
```
Comment (rakeshPRaghu) [inline website/src/components/Metadata/index.tsx:8]: we can remove the extra line

File: website/src/components/Metadata/index.tsx [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+    description: string;
+    url: string;
+    image: string;
+    type?: string;
+    isPost: boolean;
+    tags?: [string];
+    author?: Author;
+}
+
+
```
Comment (rakeshPRaghu) [inline website/src/components/Metadata/index.tsx:20]: same here

File: website/src/components/Metadata/index.tsx [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+    const site = {
+        logo: `${getHostOrigin}/images/png/logo.png`,
+        siteUrl: `${getHostOrigin}`
+    }
+    let imageObj = {
+        src: image,
+        shareImageWidth: '1000px',
+        shareImageHeight: '523px'
+    };
+
```
Comment (rakeshPRaghu) [inline website/src/components/Metadata/index.tsx:32]: same here

File: website/src/components/Metadata/index.tsx [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+    const defaultConfig = {
+        title: `OpenEBS - Container Attached Storage`,
+        description: `OpenEBS is an open source storage platform that provides persistent and containerized block storage for DevOps and
+        container environments.`,
+        url: site.siteUrl,
+        image: site.logo,
+        shareImageWidth: imageObj.shareImageWidth,
+        shareImageHeight: imageObj.shareImageHeight,
+    }
+
```
Comment (rakeshPRaghu) [inline website/src/components/Metadata/index.tsx:42]: same here

File: website/src/components/Metadata/index.tsx [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+    const jsonLd = isPost ? articleSchema({title, description, url, image: imageObj, author, tags, site }) : websiteSchema( {title, description, url, image: imageObj, type , site });
+    return(
```
Comment (rakeshPRaghu) [inline website/src/components/Metadata/index.tsx:44]: we can add an extra lin here!

File: website/src/components/Metadata/index.tsx [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+            {/** Twitter */}
+            <meta name="twitter:card" content="summary_large_image" />
+            <meta name="twitter:creator" content="@openebs" />
+            <meta name="twitter:site" content="https://twitter.com/openebs" />
+            <meta name="twitter:title" content={title || defaultConfig.title} />
+            <meta name="twitter:description" content={description || defaultConfig.description} />
+            <meta name="twitter:image" content={image || defaultConfig.image} />
+
+            {isPost && tags && tags.map((keyword, i) => (
+                    <meta property="article:tag" content={keyword} key={i} />
```
Comment (rakeshPRaghu) [inline website/src/components/Metadata/index.tsx:68]: the spacing!

Comment (rakeshPRaghu) [reply website/src/components/Metadata/index.tsx:68]: keep one tab space

File: website/src/resources/seo.json [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+        "title": "Open Source Community for OpenEBS - (Kubernetes Container Attached Storage)",
+        "description": "The only 2,000-strong member community devoted to building better Kubernetes storage. We’d love to have you, too! Join us on Slack or contribute directly on GitHub!",
+        "image": "/images/seo/openebs.png",
+        "url": "/community"
+      },
+      "commericalsupport": {
+        "title": "Commerical support for OpenEBS - (Kubernetes Container Attached Storage)",
+        "description": "List of companies providing commercial products and support around OpenEBS",
+        "image": "/images/seo/openebs.png",
+        "url": "/commerical-support"
```
Comment (Pallavi-PH) [inline website/src/resources/seo.json:26]:
```suggestion
        "url": "/commercial-support"
```

File: website/src/components/Metadata/index.tsx [hunk 10]
```diff
@@ -0,0 +1,82 @@
...
+export const Metadata: React.FC<MetadataProps> = ({ title, description, url, isPost , image, tags, author, type }) => {
+    
+    const site = {
+        logo: `${getHostOrigin}/images/png/logo.png`,
+        siteUrl: `${getHostOrigin}`
+    }
+    let imageObj = {
+        src: image,
+        shareImageWidth: '1000px',
+        shareImageHeight: '523px'
```
Comment (Pallavi-PH) [inline website/src/components/Metadata/index.tsx:30]: Recommended image size is 1200 x 630. Can we use the same? https://developers.facebook.com/docs/sharing/webmasters/images/

File: website/src/pages/BlogPage/index.tsx [hunk 11]
```diff
@@ -121,13 +124,26 @@ const BlogPage: React.FC = () => {
 
   return (
     <>
+    {currentBlogDetails.id && (
+      <Metadata 
+        title={currentBlogDetails?.title} 
+        description={currentBlogDetails?.excerpt} 
+        url={currentLocation} 
+        image={`${getHostOrigin}/images${currentPathname}.png` || `${getHostOrigin}/images/blog/defaultImage.png`}
```
Comment (Pallavi-PH) [inline website/src/pages/BlogPage/index.tsx:132]:
```suggestion
        image={`${getHostOrigin}/images/${currentPathname}.png` || `${getHostOrigin}/images/blog/defaultImage.png`} 
```

Comment (IsAmrish) [reply website/src/pages/BlogPage/index.tsx]: `window.location.pathname` provides `trailing slash`

Comment (Pallavi-PH) [reply website/src/pages/BlogPage/index.tsx]: Ok

**Review by AVRahul — APPROVED** [review]

**Review by Pallavi-PH — APPROVED** [review]

**Review by rakeshPRaghu — APPROVED** [review]
lgtm, thanks @IsAmrish

**Review by satyapriyamishra222 — APPROVED** [review]

**Pull request merged**